astroplan/skycalc.py

```diff
@@ -0,0 +1,179 @@
+            default = [500.0, 510.0, 520.0, 530.0, 540.0, 550.0]
+        lsf_type : {'none', 'Gaussian', 'Boxcar'}. default = 'none'
+            Line spread function type for convolution.
+        lsf_gauss_fwhm : float. default = 5.0
+            Gaussian full-width half-max for line spread function wavelength bins.
+        lsf_boxcar_fwhm : float. default = 5.0
+            Boxcar full-width half-max for line spread function wavelength bins.
+        observatory : {'paranal', 'lasilla', 'armazones'}. default = 'paranal'
+            Observatory where observation takes place.
+
+        Returns
+        -------
+        trans_waves, transmission: tuple of arrays of floats
+            'trans_waves' is an array of wavelengths in angstroms (float),
+            'transmission' is an array of fractional atmospheric
+            transmittance (float).
+
+        """
+        params = locals()
+
+        if params['observatory'] == 'lasilla':
+            params['observatory'] = '2400'
+        elif params['observatory'] == 'paranal':
+            params['observatory'] = '2640'
+        elif (params['observatory'] == '3060m' or
+              params['observatory'] == 'armazones'):
+            params['observatory'] = '3060'
+        else:
+            raise ValueError('Wrong Observatory name, please refer to the '
+                             'skycalc_cli documentation.')
+
+        # import packages not required by astroplan
+        import os
+        import json
+        import requests
+
+        # Use the bit from skycalc_cli which queries from the SkyCalc Sky Model
+        server = 'http://etimecalret-001.eso.org'
+        url = server + '/observing/etc/api/skycalc'
+
+        response = requests.post(url, data=json.dumps(params))
+        results = json.loads(response.text)
+
+        status = results['status']
+        tmpdir = results['tmpdir']
+        tmpurl = server + '/observing/etc/tmp/' + tmpdir + '/skytable.fits'
+
+        if status == 'success':
+            try:
+                response = requests.get(tmpurl, stream=True)
+                data = response.content
+            except requests.exceptions.RequestException as e:
+                print(e, 'could not retrieve FITS data from server')
+        else:
+            raise SkyCalcError('HTML request failed. A custom parameter you ' +
+                               'set is most likely not accepted by the SkyCalc' +
+                               ' Calculator. json.loads() returns:' +
+                               ' {}'.format(results))
+
+        # Create a temporary file to write the binary results to
+        tmp_data_file = './tmp_skycalc_data.fits'
+
+        with open(tmp_data_file, 'wb') as f:
+            f.write(data)
+
+        hdu = fits.open(tmp_data_file)
+        trans_waves = hdu[1].data["LAM"] * u.um  # wavelengths
+        transmission = hdu[1].data["TRANS"]
+
+        # Delete the file after reading from it
+        os.remove(tmp_data_file)
```

Comment:
I wonder if `astroplan` has some sort of caching mechanism so that calling this function repeatedly will not result in a network call if a previously downloaded file exists.

Reply:
@bmorris3? I don't know anything about how caching works in python (or otherwise.. 😬)

Reply:
astroplan doesn't do any caching yet, but we can imagine making a cache specifically for this kind of query.